### DISCUSSION TITLE
How to make a informative download progress bar?

### DISCUSSION THREAD

**Discussion by liblaf** [charmbracelet/bubbles#319] [Q&A] [answered]
How can I make a download progress bar with download speed, total file size, estimated remaining time and so on, like this demo ⬇️

![demo](https://user-images.githubusercontent.com/30631553/216933313-210ddc4e-8618-4bb0-b315-078618ac35b7.gif)

**Comment by bashbunni** [comment, marked as answer]
The code is linked in that same example folder. You can essentially use p.Send to send updated values to your bubble tea program. In this case, there's a function that is constantly receiving the progress updates and sends that to our bubble tea program with p.Send. Does that answer the question? If not, I can try again :)

**Reply by bashbunni** [reply]
Here's the link to the example: https://github.com/charmbracelet/bubbletea/tree/master/examples/progress-download

**Reply by liblaf** [reply]
> The code is linked in that same example folder. You can essentially use p.Send to send updated values to your bubble tea program. In this case, there's a function that is constantly receiving the progress updates and sends that to our bubble tea program with p.Send. Does that answer the question? If not, I can try again :)

I've read that example and it answers part of the question. I'm more curious about how to display information such as elapsed time, estimated remaining time, and file size.

**Reply by bashbunni** [reply]
Oh gotcha, thanks for clarifying. I don't have time to dive into solving the time calculations at the moment, but you can pass any and all data from the response to your bubble tea program. Here I've updated the example to also include the download progress in bytes alongside the progress bar. I hope that helps! 

https://github.com/bashbunni/answering-issues/tree/main/progress

**Reply by liblaf** [reply]
> Oh gotcha, thanks for clarifying. I don't have time to dive into solving the time calculations at the moment, but you can pass any and all data from the response to your bubble tea program. Here I've updated the example to also include the download progress in bytes alongside the progress bar. I hope that helps!
> 
> https://github.com/bashbunni/answering-issues/tree/main/progress

Thanks! This is very helpful.
